Code/GraphMol/Kekulize.cpp

```diff
@@ -512,17 +512,41 @@ void Kekulize(RWMol &mol, bool markAtomsBonds, unsigned int maxBackTracks) {
   //       - getBondType return aromatic
   // - all aromatic atoms return true for "getIsAromatic"
 
-  // first find the all the simple rings in the molecule
+  // first find the all the simple rings in the molecule that are not
+  // completely composed of dummy atoms
   VECT_INT_VECT arings;
-  if (mol.getRingInfo()->isInitialized()) {
-    arings = mol.getRingInfo()->atomRings();
+  boost::dynamic_bitset<> dummyAts(mol.getNumAtoms());
+  for (ROMol::AtomIterator atit = mol.beginAtoms(); atit != mol.endAtoms();
```

Comment:
I think we can make the assumption that rings are initialized, mainly since kekulize is called after SANITIZE_KEKULIZE.

We might gain an optimization if we assign allrings before dummyAts and then only add dummy atoms  to dummyAts that are indeed in rings.